src/DeviceManager.Api/Configuration/EntityFrameworkConfiguration.cs

```diff
@@ -17,13 +21,44 @@ public static class EntityFrameworkConfiguration
         /// <param name="configuration">The configuration.</param>
         public static void ConfigureService(IServiceCollection services, IConfigurationRoot configuration)
         {
-            string connectionString = configuration.GetConnectionString("DefaultConnection");
+            string connectionString = configuration.GetConnectionString(Constants.DefaultConnection);
+
+            // Database connection settings
+            var connectionOptions = services.BuildServiceProvider().GetRequiredService<IOptions<ConnectionSettings>>();
 
             // Entity framework configuration
-            services.AddDbContext<DeviceContext>(options =>
-                options.UseSqlServer(connectionString));
+            services.AddEntityFrameworkNpgsql().AddDbContext<DeviceContext>(options =>
```

Comment:
Very nice. Thanks. Just wandering, why you choose PostgreSQL ? There are other powerful nosql like MongoDb, CouchDB etc.

Reply:
I will test with other no sql databases also. Then you can accept the pull request.